crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -238,6 +228,18 @@ impl BaseClient {
             }
         }
 
+        // Process receipts now we have rooms
+        if let Some(receipts) = receipt.as_ref() {
```

Comment:
Assuming `receipt` is `Option<Something>`, this will work too

```suggestion
        if let Some(receipts) = &receipt {
```